### ISSUE TITLE
WooCommerce notifications not sent from sub-site on WPMS (mailer: SMTP)

### ISSUE THREAD

**Issue by jonasskafte** [awesomemotive/wp-mail-smtp#117]
## Plugin version
- 1.2.2

## Current Behavior
- Child-site won't send WooCommerce notification emails via SMTP (Office 365).
- Works fine with standard WordPress PHP mailer.
- Works fine on mother-site of multisite installation.
- Works fine when sending non-WooCommerce WordPress emails.
- Not consistent behaviour – a few times the emails has actually gone through, but it has been very rare.

## Expected Behaviour
- I guess I was just expecting that it would work like on the mother-site 🤔 

## Possible Solution
- I've got no idea – sorry! -but I hope you do 😃 
- Or actually, maybe this issue could be solved with a global SMTP solution, like [this one](https://github.com/rohmann/global-smtp)?

## Screenshots (if appropriate)
- I haven't made screenshots, but here's a video where I show you the issue in action. Hope that's useful: [https://vimeo.com/250663461](https://vimeo.com/250663461) _(removed the p from message)_

**Comment by slaFFik** [comment]
Thanks for the report!

Regarding Vimeo link: "Sorry, we couldn’t find that page" - so I can't see it.
Plugin already supports constants. You can see them all in `Options.php` file, in `get_const_value()` method.

This issue might be related to #90 and #100.

**Comment by jonasskafte** [comment]
Hi there @slaFFik,

Thanks for looking into this and for dealing with my enquiry so thoroughly!

You should now be able to see the video – sorry, the privacy settings hadn't been saved properly. My bad!

I'm glad to hear that you have a potential solution for this and thanks for letting me know about constants – however I don't know how to make use of this myself, so I guess I'll just wait and hope for a solution in the next release of the plugin.

Many thanks!

**Comment by slaFFik** [comment]
Still error: "This video is private".

**Comment by slaFFik** [comment]
Ok, I see now, thanks.

**Comment by jonasskafte** [comment]
Thanks! I'm glad it was useful 😃 

- WooCommerce is _not_ network activated, but activated on the sub-site and parent-site separately. Should I have activated it on network level?
- Yes, it is a subdomain with domain mapping – I just used the inbuilt WordPress functionality to do this and not a plugin.

**Comment by slaFFik** [comment]
1. I just need to know your setup.
2. Ok, great. Thanks.

**Comment by pieterbeulque** [comment]
Same setup (WooCommerce not network-activated + domain mapping) and also having trouble with emails not showing up. WooCommerce notifications, mostly.

**Comment by slaFFik** [comment]
@pieterbeulque
Mailer: SMTP too?

**Comment by pieterbeulque** [comment]
```php
define( 'WPMS_ON', true );
define( 'WPMS_MAIL_FROM', 'redacted' );
define( 'WPMS_MAIL_FROM_NAME', 'Mr. Henry' );
define( 'WPMS_MAILER', 'smtp' ); // Possible values 'smtp', 'mail', or 'sendmail'
define( 'WPMS_SET_RETURN_PATH', false ); // Sets $phpmailer->Sender if true
define( 'WPMS_SMTP_HOST', 'maildev' ); // The SMTP mail host
define( 'WPMS_SMTP_PORT', 25 ); // The SMTP server port number
define( 'WPMS_SSL', '' ); // Possible values '', 'ssl', 'tls' - note TLS is not STARTTLS
define( 'WPMS_SMTP_AUTH', false ); // True turns on SMTP authentication, false turns it off
define( 'WPMS_SMTP_USER', '' ); // SMTP authentication username, only used if WPMS_SMTP_AUTH is true
define( 'WPMS_SMTP_PASS', '' ); // SMTP authentication password, only used if WPMS_SMTP_AUTH is true
```

Copied from our `wp-config.php`.

If you want me to check anything out, let me know if I can help 🙂

**Comment by jonasskafte** [comment]
Hi there @slaFFik,

Hope you've had a nice weekend.

I was just wondering if you could give us an indicator as to when approximately you expect to have version 1.3.0 of the plugin out? -just good to know, so that we can consider other solutions in the meantime.

Many thanks!

**Comment by jonasskafte** [comment]
Hi @slaFFik,

Hope you are well.

As it seems it might be a while before 1.3.0 will be out, do you think there's any chance that this could be fixed in the meantime for a version 1.2.4?

Sorry to ask, but it's getting really critical for me to find a solution as soon as possible.

Kind regards,

**Comment by slaFFik** [comment]
Closing this in favor of #100.

**Comment by jonasskafte** [comment]
Hi there @slaFFik 

Thanks again for looking into this.

I was wondering – now that you've 'closed' this issue (or spread it into other issues) are you still planning to fix the problem? I am thinking – in issue #100 ('Better Multisite Support') the focus seems to be on making the plugin better generally for multisite installations (add more features), however, I think there's a more urgent matter, which is to simply make the plugin function for sub-sites. At the moment no emails are sent from sub-sites at all. Is that a fix that could perhaps be kept in milestone 1.3.0, as it's a matter of urgency, and then the new features could be saved for 1.4.0?

Thanks!
